apps/classroomio-com/src/routes/tools/progress/components/Avatar.svelte

```diff
@@ -0,0 +1,91 @@
+<script lang="ts">
+  import { openAvatar, closeAvatarModal, htmlBody, type Source } from './store';
+
+  let isAddIcon = false;
+  let hoveredIndex: number | null = null;
+
+  let avatars = [
+    { src: 'avatar_a' },
+    { src: 'avatar_u' },
+    { src: 'avatar_b' },
+    { src: 'avatar_c' },
+    { src: 'avatar_d' },
+    { src: 'avatar_e' },
+    { src: 'avatar_g' },
+    { src: 'avatar_h' },
+    { src: 'avatar_i' },
+    { src: 'avatar_j' },
+    { src: 'avatar_k' },
+    { src: 'avatar_l' },
+    { src: 'avatar_m' },
+    { src: 'avatar_n' },
+    { src: 'avatar_o' },
+    { src: 'avatar_p' },
+    { src: 'avatar_q' },
+    { src: 'avatar_r' },
+    { src: 'avatar_s' },
+    { src: 'avatar_t' }
+  ];
+
+  function selectAvatar(avatar: Source) {
+    $htmlBody.avatar = avatar.src;
+  }
+</script>
+
+{#if $openAvatar.open}
+  <div class="relative">
+    <!-- black background -->
+    <div class="fixed z-[3000] top-0 left-0 w-full h-full bg-black opacity-[0.7]"></div>
+    <div
+      class="fixed top-[20%] left-10 md:left-[30%] z-[3001] w-[80%] md:w-[43%] bg-white rounded-md mx-auto py-6 px-7"
```

Comment:
@tunny17 I noticed you are recreating the logic of modal in all 3 modals which is bad.

I mentioned earlier on our call that you should look at how the Modal component was created in our dashboard project and use something similar.

In this case we might not need to implement a reusable navbar within a modal, just the container any of the 3 modals can reuse the logic.

Create one component in `src/lib/Modal/index.svelte`

Then in the Fullscreen, Mood, Background to use the `Modal` component and then add the content for the modal like:


```
// BackroundModal.svelte
<Modal>
  {/* Add the content of the modal for Background*/}
</Modal>

// MoodModal.svelte
<Modal>
  {/* Add the content of the modal for Mood*/}
</Modal>

//etc
```